### PR TITLE
fix: `FORM_SUBMITTED` webhook payload change and support for Team Webhooks with it

### LINKED ISSUE #9037: [CAL-1735] FORM_SUBMITTED webhook doesn't have formId and other details of forms
It just has the responses. Like EventType webhooks it would make sense to provide form details as well.

Also, add the routing action taken for the submission.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1735](https://linear.app/calcom/issue/CAL-1735/form-submitted-webhook-doesnt-have-formid-and-other-details-of-forms)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

- `FORM_SUBMITTED` webhook is much more usable now. Earlier it just had the responses in it, now it has more details about the webhook itself and the associated form
- Team Webhooks can now be created for FORM_SUBMITTED webhook. A team Routing Form response would fire that team's webhook only.

Fixes #9037

## Old Webhook Body format

```json
{
  "Multi": ["3"],
  "numField": 2
}
```

## New Webhook Body format
```json
{
  "triggerEvent": "FORM_SUBMITTED",
  "createdAt": "2023-08-28T08:34:41.823Z",
  "payload": {
    "formId": "cdf81606-0bac-4f65-b588-9407e43015af",
    "formName": "Engineering Form",
    "teamId": 27,
    "responses": {
      "Multi": [
        "3"
      ],
      "numField": 2
    }
  }
}
```

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
- Create a non-team(i.e User) webhook for FORM_SUBMITTED and submit a user Routing Form. Verify that the webhook is fired with new payload now.
- Create a team webhook for FORM_SUBMITTED(which wasn't allowed earlier). Verify that the webhook is fired for the Team's Routing Form submission. 
	- Verify that it doesn't fire for the User Routing Form submission
	- Verify that it doesn't fire for any other Team's Routing Form Submission

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/playwright/webhook.e2e.ts [hunk 1]
```diff
@@ -9,6 +10,9 @@ import {
   gotoRoutingLink,
 } from "./lib/testUtils";
 
+// remove dynamic properties that differs depending on where you run the tests
+const dynamic = "[redacted/dynamic]";
```
Comment (hariombalhara) [inline apps/web/playwright/webhook.e2e.ts:14]: Maintained `dynamic` at single place

File: apps/web/playwright/webhook.e2e.ts [hunk 2]
```diff
@@ -391,26 +389,14 @@ test.describe("BOOKING_REQUESTED", async () => {
...
-    await page.locator('input[name="name"]').fill(teamName);
-    await page.locator('input[name="slug"]').fill(teamName);
-    await page.locator('button[type="submit"]').click();
-
-    await page.locator("text=Publish team").click();
-    await page.waitForURL(/\/settings\/teams\/(\d+)\/profile$/i);
-
-    await page.waitForLoadState("networkidle");
-
-    await page.waitForLoadState("networkidle");
```
Comment (hariombalhara) [inline apps/web/playwright/webhook.e2e.ts:413]: Speed up test by creating team from query itself.

File: apps/web/playwright/webhook.e2e.ts [hunk 3]
```diff
@@ -391,26 +389,14 @@ test.describe("BOOKING_REQUESTED", async () => {
 });
 
 test.describe("FORM_SUBMITTED", async () => {
-  test("can submit a form and get a submission event", async ({ page, users }) => {
+  test("on submitting user form, triggers user webhook", async ({ page, users, routingForms }) => {
     const webhookReceiver = createHttpServer();
-    const user = await users.create();
+    const user = await users.create(null, {
+      hasTeam: true,
```
Comment (hariombalhara) [inline apps/web/playwright/webhook.e2e.ts:395]: replacement of team creation flow.

File: apps/web/playwright/webhook.e2e.ts [hunk 4]
```diff
@@ -422,23 +408,121 @@ test.describe("FORM_SUBMITTED", async () => {
...
+      teamId: null,
+      fields: [
+        {
+          type: "text",
+          label: "Name",
+          identifier: "name",
+          required: true,
+        },
+      ],
+    });
```
Comment (hariombalhara) [inline apps/web/playwright/webhook.e2e.ts:424]: Create form using fixture

File: apps/web/playwright/webhook.e2e.ts [hunk 5]
```diff
@@ -422,23 +408,121 @@ test.describe("FORM_SUBMITTED", async () => {
...
     page.click('button[type="submit"]');
 
     await waitFor(() => {
       expect(webhookReceiver.requestList.length).toBe(1);
     });
+    const [request] = webhookReceiver.requestList;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const body = request.body as any;
+    body.createdAt = dynamic;
+    expect(body).toEqual({
```
Comment (hariombalhara) [inline apps/web/playwright/webhook.e2e.ts:438]: Verify the payload as well, earlier were were just checking if the webhook was sent.

File: apps/web/playwright/webhook.e2e.ts [hunk 6]
```diff
@@ -422,23 +408,121 @@ test.describe("FORM_SUBMITTED", async () => {
...
+      },
+      responses: {
+        name: "John Doe",
+      },
+    });
+
+    webhookReceiver.close();
+  });
+
+  test.only("on submitting team form, triggers team webhook", async ({ page, users, routingForms }) => {
```
Comment (hariombalhara) [inline apps/web/playwright/webhook.e2e.ts:459]: Added a webhook test form Team Routing Forms.

File: apps/web/playwright/webhook.e2e.ts [hunk 7]
```diff
@@ -422,23 +408,121 @@ test.describe("FORM_SUBMITTED", async () => {
...
+
+    // Add webhook
+    await page.fill('[name="subscriberUrl"]', webhookReceiver.url);
+    await page.fill('[name="secret"]', "secret");
+    await page.click("[type=submit]");
+
+    const form = await routingForms.create({
+      name: "Test Form",
+      userId: user.id,
+      teamId: teamId,
```
Comment (hariombalhara) [inline apps/web/playwright/webhook.e2e.ts:477]: create RoutingForm for the same team for which webhook was created.

File: packages/app-store/routing-forms/trpc/utils.ts [hunk 8]
```diff
@@ -12,32 +12,51 @@ export async function onFormSubmission(
...
+      value: fieldResponse.value,
+    };
   }
 
   const subscriberOptions = {
-    userId: form.user.id,
     triggerEvent: WebhookTriggerEvents.FORM_SUBMITTED,
-    // When team routing forms are implemented, we need to make sure to add the teamId here
-    teamId: null,
+    ...getWebhookTargetEntity(form),
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/trpc/utils.ts:34]: Abstracted out this logic.

File: packages/features/webhooks/components/WebhookForm.tsx [hunk 9]
```diff
@@ -57,15 +57,13 @@ const WebhookForm = (props: {
...
   onCancel?: () => void;
-  noRoutingFormTriggers: boolean;
 }) => {
   const { apps = [] } = props;
   const { t } = useLocale();
 
   const triggerOptions = [...WEBHOOK_TRIGGER_EVENTS_GROUPED_BY_APP_V2["core"]];
   if (apps) {
     for (const app of apps) {
-      if (app === "routing-forms" && props.noRoutingFormTriggers) continue;
```
Comment (hariombalhara) [inline packages/features/webhooks/components/WebhookForm.tsx:68]: Now, Routing Form Webhooks are also supported for Teams

File: packages/features/webhooks/lib/sendPayload.ts [hunk 10]
```diff
@@ -126,24 +126,37 @@ const sendPayload = async (
...
+  webhook,
+  data,
+  rootData,
+}: {
+  secretKey: string | null;
+  triggerEvent: string;
+  createdAt: string;
+  webhook: Pick<Webhook, "subscriberUrl" | "appId" | "payloadTemplate">;
+  data: Record<string, unknown>;
+  rootData?: Record<string, unknown>;
```
Comment (hariombalhara) [inline packages/features/webhooks/lib/sendPayload.ts:145]: A new rootData prop to be able to support adding props at root level.

File: packages/app-store/typeform/playwright/tests/basic.e2e.ts [hunk 11]
```diff
@@ -4,7 +4,7 @@ import { expect } from "@playwright/test";
 import {
   addForm as addRoutingForm,
   addOneFieldAndDescriptionAndSaveForm,
-} from "@calcom/app-store/routing-forms/playwright/tests/basic.e2e";
+} from "@calcom/app-store/routing-forms/playwright/tests/testUtils";
```
Comment (hariombalhara) [inline packages/app-store/typeform/playwright/tests/basic.e2e.ts:7]: A small refactor required before upgrading playwright. There were more issues so we can tackle them separately.

**Review by CarinaWolli — APPROVED** [review]
Tested and works, code looks good as well 👍

**Pull request merged**